saleor/graphql/attribute/mutations.py

```diff
@@ -350,21 +372,20 @@ def _clean_attribute_settings(cls, instance, cleaned_input):
         Ensure that any invalid operations will be not performed.
         """
         attribute_input_type = cleaned_input.get("input_type") or instance.input_type
-        if attribute_input_type not in [
-            AttributeInputType.FILE,
-            AttributeInputType.REFERENCE,
-        ]:
-            return
         errors = {}
-        for field in [
-            "filterable_in_storefront",
-            "filterable_in_dashboard",
-            "available_in_grid",
-            "storefront_search_position",
-        ]:
-            if cleaned_input.get(field):
+        if attribute_input_type == AttributeInputType.NUMERIC:
+            if not cleaned_input.get("unit"):
+                errors["unit"] = ValidationError(
+                    "The unit value must be set for numeric attribute.",
+                    code=AttributeErrorCode.REQUIRED.value,
+                )
```

Comment:
I think units should be optional, e.g. when you wanted to create an attribute for product rating that would be just a number.